coil/src/main/java/dev/chrisbanes/accompanist/coil/Coil.kt

```diff
@@ -215,26 +173,123 @@ fun CoilImage(
             lastRequestedSize.value = requestSize
         }
 
-        if (painter == null) {
-            // If we don't have a result painter, we add a Box...
-            Box {
-                // If we don't have a result yet, we can show the loading content
-                // (if not null)
-                if (result == null && loading != null) {
-                    loading()
+        when (r) {
```

Comment:
Why can't you use `result` instead of `r`?

Reply:
It's a delegate property so the compiler can't guarantee the value won't change.